index.js

```diff
@@ -1 +1,105 @@
 // implement your API here
+const express = require("express");
+const Db = require("./data/db");
+
+const server = express();
+
+server.use(express.json());
+
+server.get("/", (req, res) => {
+  res.send("Monday July 8 WebAPI Challenge");
+});
+
+server.get("/api/users", function(req, res) {
+  Db.find()
+    .then(users => {
+      res.status(200).json(users);
+    })
+    .catch(error => {
+      res
+        .status(500)
+        .json({ error: "The users information could not be retrieved." });
+    });
+});
+
+server.get("/api/users/:id", function(req, res) {
+  const { id } = req.params;
+  console.log(id);
+  Db.findById(id)
+    .then(user => {
+      res.status(200).json(user);
+    })
+    .catch(error => {
+      res
+        .status(404)
+        .json({ message: "The user with the specified ID does not exist." });
+    });
+});
+
+server.post("/api/users", (req, res) => {
+  const userInfo = req.body;
+  console.log(userInfo);
+
+  Db.insert(userInfo)
+    .then(user => {
+      if (user) {
+        res.status(204).send(user);
+      } else {
+        res
+          .status(400)
+          .json({ errorMessage: "Please provide name and bio for the user." });
+      }
+    })
+    .catch(error => {
+      res.status(500).json({
+        error: "There was an error while saving the user to the database"
+      });
+    });
+});
+
+server.delete("/api/users/:id", (req, res) => {
+  const { id } = req.params;
+
+  Db.remove(id)
+    .then(deleted => {
+      if (deleted === 1) {
+        res.status(204).end();
```

Comment:
They don't specify what to return from a successful delete, but we should probably return something so the user knows it has been successful. Status 204 means no content is returned, but there might be useful information on the headers. Check out this link: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204 . So you could maybe do something with that if you would like, but it's not something I would be familiar with right away. Instead, I like to return a 200 status code, and I'll send back the list of the new users. Do this by doing another Db.find(), and return 200
```suggestion
        Db.find().then(users => res.status(200).json(users));
```